### ISSUE TITLE
Next release discussions

### ISSUE THREAD

**Issue by thothix** [thothix/guayadeque#33]
As always we have here a path to next release discussions (it can be 0.6.3 or 0.7.0, who knows!?, I don't have a release plan yet!).

Feel free to open a new issue to specific subjects.

**Comment by filochard** [comment]
Hi Tiago
For the next release number I would prefer 0.6.3  unless there are major changes in the code

Besides this I discovered (too late) a little typo in Catalan translation
no emergency for 0.6.2 but it may be corrected for the next

in /po/ca_ES/guayadeque.po  
  I forgot a l·l needed for a correct pronunciation 

#: misc/Accelerators.cpp:164 ui/MainFrame.cpp:1513
msgid "Smart Play"
msgstr "Reproducció intelligent"
should be msgstr "Reproducció intel·ligent"

**Comment by filochard** [comment]
Thanks for the correction of the typo

About the changes in dev branch concerning .aac files : 
I don't use the .aac extension but, a long time ago, I had converted all my  .aac files into .m4a files so that they can be tagged...
This workaround was systematically applied to every new .aac file of my collection

To test with this last commit I'm gonna import an .aac file ;-)

**Comment by thothix** [comment]
> Thanks for the correction of the typo
> 
> About the changes in dev branch concerning .aac files : I don't use the .aac extension but, a long time ago, I had converted all my .aac files into .m4a files so that they can be tagged... This workaround was systematically applied to every new .aac file of my collection
> 
> To test with this last commit I'm gonna import an .aac file ;-)

Hi!
Thank you for testing this.

I don't have AAC files in my library as well. I Added a couple files to test it.
It's important to import some of them because this can lead to scan crashes. 

I got an issue with .aac files in 0.5.0, I had only 1 aac file and I wasn't able to import my library because it was crashing. The solution was to remove the .aac file from the library (until the fix in 0.5.0 beta 2)
If it doesn't crash the library scan is already a good thing! 
I also tested the "Edit Songs" in mediaviewer and it worked well.

**Comment by filochard** [comment]
I just imported a mp4 file converted it in m4a and in aac (same codec indeed)
Result :
the m4a file can correctly be tagged by Guayadeque (title, artist, lyrics)
the aac file can't be tagged by Guayadeque  (but it can be tagged by kid3)

My old workaround has prevented me to discover this bug again for years (I knew that I shouldn't use .aac files)

NB I haven't imported yet this .aac file into my library
I only opened it with Guayadeque inside my file Brower

I will backup my /home/myname/.guayadeque before trying to add it in the library

**Comment by filochard** [comment]
Last news trying to import an aac file in Guayadeque 0.6.2
I copy it into my music directory update the library : it's not found
no crash !
I try to import it with the import panel
But I can't validate to import this file...

Same test with an m4a version of the same file
It can be imported into the right folder
I can find the file by looking for its name in the music tab
Then it can be tagged (artist, title, album name, Lyrics)

I understand why I never installed aac files in my Music directory : they can't be found !

I will try later to install a git with the last commits

**Comment by thothix** [comment]
> Last news trying to import an aac file in Guayadeque 0.6.2 I copy it into my music directory update the library : it's not found no crash ! I try to import it with the import panel But I can't validate to import this file...
> 
> Same test with an m4a version of the same file It can be imported into the right folder I can find the file by looking for its name in the music tab Then it can be tagged (artist, title, album name, Lyrics)
> 
> I understand why I never installed aac files in my Music directory : they can't be found !
> 
> I will try later to install a git with the last commits

Did you import it in 0.6.2?
.AAC files are not imported by Guayadeque from 0.5.0 to 0.6.2 and before 0.5.0 they crashed the scan (I don't know if it worked in older versions before 0.4.7).
They are imported now in 0.6.3-beta.

**Comment by filochard** [comment]
> > Last news trying to import an aac file in Guayadeque 0.6.2 I copy it into my music directory update the library : it's not found no crash ! I try to import it with the import panel But I can't validate to import this file...
> > Same test with an m4a version of the same file It can be imported into the right folder I can find the file by looking for its name in the music tab Then it can be tagged (artist, title, album name, Lyrics)
> > I understand why I never installed aac files in my Music directory : they can't be found !
> > I will try later to install a git with the last commits
> 
> Did you import it in 0.6.2? .AAC files are not imported by Guayadeque from 0.5.0 to 0.6.2 and before 0.5.0 they crashed the scan (I don't know if it worked in older versions before 0.4.7). They are imported now in 0.6.3-beta.

I did try to import with 0.6.2
Didn't try to build 0.6.3-beta yet

**Comment by filochard** [comment]
Hi Tiago
I built and installed 0.6.3-beta  (commit 914ae77)
Now it can import a non tagged .aac file  using simply its name  in the import panel and validating 
But it doesn't import simply the file inside the chosen directory  /Music /m4a/filename.aac
but with its whole previous path /Musi /m4a/Inconnu/Inconnu/dwhelper/filename.aac
as if th first Inconnu was the album artist name, then the second Inconnu was the name, and dwelper the album name

After this Guayadeque can't find this file to read or tag it : I need to show the browser panel to find it
I can play it then and tag it 
not so simple 
and these tags are not standard for aac

If the aac is already tagged before importing it the tag is lost when imported....

Not so good yet
I revert to 0.6.2

PS I had never used the import file panel
I always simply copy files with my own file browser (Dolphin) from a directory (where it stays) 
to the directory folder or subfolder where I want to drop it

**Comment by filochard** [comment]
[Goksel_Baktagir-Lena_Chamamyan_Unutmak_Mumkun_mu.zip](https://github.com/user-attachments/files/18156938/Goksel_Baktagir-Lena_Chamamyan_Unutmak_Mumkun_mu.zip)
Here is an aac file for your tests

**Comment by thothix** [comment]
> > > Last news trying to import an aac file in Guayadeque 0.6.2 I copy it into my music directory update the library : it's not found no crash ! I try to import it with the import panel But I can't validate to import this file...
> > > Same test with an m4a version of the same file It can be imported into the right folder I can find the file by looking for its name in the music tab Then it can be tagged (artist, title, album name, Lyrics)
> > > I understand why I never installed aac files in my Music directory : they can't be found !
> > > I will try later to install a git with the last commits
> > 
> > 
> > Did you import it in 0.6.2? .AAC files are not imported by Guayadeque from 0.5.0 to 0.6.2 and before 0.5.0 they crashed the scan (I don't know if it worked in older versions before 0.4.7). They are imported now in 0.6.3-beta.
> 
> I did try to import with 0.6.2 Didn't try to build 0.6.3-beta yet

Ok. I got some issues testing.
Easytag and Puddletag don't support AAC tagging. My Kid3 is crashing at start and I ended running Kid3 in a VM! 
And I figured out that Guayadeque is not updating the file tags although tag info appears in the songs editor, it is only in the database, the aac file is not updated!

**Comment by thothix** [comment]
> Hi Tiago I built and installed 0.6.3-beta (commit [914ae77](https://github.com/thothix/guayadeque/commit/914ae775f8f3cae09d8e1a39985cf78063baa7e4)) Now it can import a non tagged .aac file using simply its name in the import panel and validating But it doesn't import simply the file inside the chosen directory /Music /m4a/filename.aac but with its whole previous path /Musi /m4a/Inconnu/Inconnu/dwhelper/filename.aac as if th first Inconnu was the album artist name, then the second Inconnu was the name, and dwelper the album name
> 
> After this Guayadeque can't find this file to read or tag it : I need to show the browser panel to find it I can play it then and tag it not so simple and these tags are not standard for aac
> 
> If the aac is already tagged before importing it the tag is lost when imported....
>

It's because the tags are only in the database and in the rescan the songs in the database are deleted to be inserted again.
but Guayadeque don't import the AAC tags from the files so the new database tags data turns empty.
 
> Not so good yet I revert to 0.6.2
> 
> PS I had never used the import file panel I always simply copy files with my own file browser (Dolphin) from a directory (where it stays) to the directory folder or subfolder where I want to drop it

What do you mean by "import file panel"? 
I'm not sure what is it. We have "Update" and "Rescan" in the library/folders/Songs

Do you mean the "Import Files" in Collection menu ? This is for copying external files, its not intended to scan the library.

**Comment by thothix** [comment]
Summarizing:

There's no way to scan AAC files tags in Guayadeque!
I tried in the code without success, the Taglib lib doesn't support AAC tags! We can only import it (with empty tags) and play it. 

I think Taglib 2 can scan it but most systems don't have it. I have Taglib 1.13 in Linux Mint (from Ubuntu 24.04) . I think Mageia has 1.12. 
I will test it in Arch Linux that has Taglib 2 already.

**Comment by filochard** [comment]
> > PS I had never used the import file panel I always simply copy files with my own file browser (Dolphin) from a directory (where it stays) to the directory folder or subfolder where I want to drop it
> 
> What do you mean by "import file panel"? I'm not sure what is it. We have "Update" and "Rescan" in the library/folders/Songs
> 
> Do you mean the "Import Files" in Collection menu ? This is for copying external files, its not intended to scan the library.

Yes that's what I meant :
Using "Import Files" in collection menu
![import-file](https://github.com/user-attachments/assets/7d2255a9-513f-4e8d-8a43-4e94586931b8)
with 0.6.2 I can import an m4a file (it appears inside the panel) and I can import it by validating
but I can't import the same file with the aac extension :  it doesn't even appear in the window

With 0.6.3-beta
I can import this acc file 

But the most surprising behaviour then is that importing a file this way functions as if the original path to the file was supposed to be : 
/album-artist-name/artist/album/file.aac
That leads to copy the file in my music directory on the drive like this : 
/Musique/m4a/Inconnu/Inconnu/dwhelper/file.aac
dwhelper being the /home/dwhelper from which the file is copied

This happens when the aac file is previously tagged or not

Nevertheless, after having imported a file, Guayadeque updates its library to add the new file (not found when it is an aac file)

NB importing a tagged (with no album-artist-name) m4a file this way, 
it is copied inside /Musique/m4a/Inconnu/file.aac
but it appears in the updated library

**Comment by filochard** [comment]
the imported tagged m4a file appearing in the auto-updated library
![library](https://github.com/user-attachments/assets/a3628ee6-3ac2-42b5-b7e6-8c4395254653)

The place where it has been copied (Guayadeque reads the tag and adds an /Inconnu/ intermediate path
 because of the missing Album-artist-name
![dolphin](https://github.com/user-attachments/assets/d9fdef7c-8618-41b6-93a2-95495ddd674a)

**Comment by filochard** [comment]
> Summarizing:
> 
> There's no way to scan AAC files tags in Guayadeque! I tried in the code without success, the Taglib lib doesn't support AAC tags! We can only import it (with empty tags) and play it.
> 
> I think Taglib 2 can scan it but most systems don't have it. I have Taglib 1.13 in Linux Mint (from Ubuntu 24.04) . I think Mageia has 1.12. I will test it in Arch Linux that has Taglib 2 already.

Yes : Mageia has Taglib 1.12.1
PS
Inside Mageia kid3 works well and can tag aac files

**Comment by thothix** [comment]
> with 0.6.2 I can import an m4a file (it appears inside the panel) and I can import it by validating but I can't import the same file with the aac extension : it doesn't even appear in the window
> 

As I said, AAC is not imported and has no support in the library in Guayadeque 0.5.0 to 0.6.2

> With 0.6.3-beta I can import this acc file
> 
> But the most surprising behaviour then is that importing a file this way functions as if the original path to the file was supposed to be : /album-artist-name/artist/album/file.aac That leads to copy the file in my music directory on the drive like this : /Musique/m4a/Inconnu/Inconnu/dwhelper/file.aac dwhelper being the /home/dwhelper from which the file is copied
> 
> This happens when the aac file is previously tagged or not
> 
> Nevertheless, after having imported a file, Guayadeque updates its library to add the new file (not found when it is an aac file)
> 
> NB importing a tagged (with no album-artist-name) m4a file this way, it is copied inside /Musique/m4a/Inconnu/file.aac but it appears in the updated library

Ok. As you, I don't use this "Import File" feature. I use the file manager to put the songs in the right location cause I have full control where to place them. I categorize my files by /genre/artist, other people can do it another way. 
I don't have time right now to test and do any change to this feature so I am going to keep it as it is by now.

**Comment by thothix** [comment]
> > Summarizing:
> > There's no way to scan AAC files tags in Guayadeque! I tried in the code without success, the Taglib lib doesn't support AAC tags! We can only import it (with empty tags) and play it.
> > I think Taglib 2 can scan it but most systems don't have it. I have Taglib 1.13 in Linux Mint (from Ubuntu 24.04) . I think Mageia has 1.12. I will test it in Arch Linux that has Taglib 2 already.
> 
> Yes : Mageia has Taglib 1.12.1 PS Inside Mageia kid3 works well and can tag aac files

Yes, kid3 support it but I saw some comments about some music players getting issues recognizing the tags written by kid3. Tags in .AAC files are written as MP3 tags by Kid3 (or any other app the support it). Easytag for instance don't support it but if you rename the file.acc to file.mp3 you are able to tag it. 
 
I tested Guayadeque 0.6.3 in Arch (taglib 2) and it is not able to tag the .aac files as well.

So I think is better to keep the support to .AAC files disabled cause if someone import it and create tags for them, the tags will be lost in the next library scan/update. 
This is a problem, it is better to have no support that half/buggy support.

**Comment by filochard** [comment]
"Import File" feature creates indeed problems if the imported file is not totally tagged : 
the chosen folder can have successive useless sub-folders.
I prefer like you copying myself the files where I want in the  Music directory of my hard drive...

About reading aac files, Guaydeque (even 0.6.2) can do this : if you choose an aac file on your hard drive with your Desktop File Browser, you may ask to open it with Guayadeque ... and it reads it. Of course it won't display any tag but you can hear it

NB taglib 2 (writing and reading aac tags) is available in Mageia Cauldron and will be present in Mageia10

When this taglib 2 is spread over all the distributions you may try to use it to build Guayadeque... not yet !

**Comment by thothix** [comment]
> "Import File" feature creates indeed problems if the imported file is not totally tagged : the chosen folder can have successive useless sub-folders. I prefer like you copying myself the files where I want in the Music directory of my hard drive...
>

Right.  I though to remove it indeed! 
Although let's keep there, maybe we can improve it someday, but it is a low priority task.
 
> About reading aac files, Guaydeque (even 0.6.2) can do this : if you choose an aac file on your hard drive with your Desktop File Browser, you may ask to open it with Guayadeque ... and it reads it. Of course it won't display any tag but you can hear it
> 

Great that it works this way but the main Guayadeque purpose is manage collections and this is the goal with AAC as well.

> NB taglib 2 (writing and reading aac tags) is available in Mageia Cauldron and will be present in Mageia10
> 
> When this taglib 2 is spread over all the distributions you may try to use it to build Guayadeque... not yet !

I did some more testing in Arch (taglib 2) and found out that tagging works.  Before I had  a problem with a malformed file.

I had an idea to make AAC support enabled in the library without get issues with tagging!
I'll disable the editing in Track Editor if tagging is not supported (disabling the fields for editing) and as a bonus we can edit it when guayadeque uses Taglib 2.

**Comment by thothix** [comment]
Change already committed to dev, beta2.

Do you have Cauldron installed? If you have it you can try to run on it.

Is there any .ISO to install Cauldron?

**Comment by filochard** [comment]
> Change already committed to dev, beta2.
> 
> Do you have Cauldron installed? If you have it you can try to run on it.
> 
> Is there any .ISO to install Cauldron?

I don't have Cauldron installed :  it's too unstable now
There is not yet any ISO to install Cauldron

**Comment by filochard** [comment]
I have just tested 0.6.3-beta2
I can use the file browser existing inside Guayadeque, select an aac file play it 
of course this aac file doesn't happear in the library

When I select the aac file playing and try to edit the tags of this track, the tag editor is grayed : that's what is wanted !
This doesn't happen for mp3 m4a ogg etc for them we can edit their tag.

**Comment by thothix** [comment]
> I have just tested 0.6.3-beta2 I can use the file browser existing inside Guayadeque, select an aac file play it of course this aac file doesn't happear in the library
> 

Didn't you try to update the library to access the AAC from mediaviewer?
You can try this,  it should be added to the library without issues.

> When I select the aac file playing and try to edit the tags of this track, the tag editor is grayed : that's what is wanted ! This doesn't happen for mp3 m4a ogg etc for them we can edit their tag.

Great!
In my tests with Taglib 2, it appears editable and I was able to edit without issues (I checked with Kid3 if the tag was updated!).

**Comment by filochard** [comment]
> > I have just tested 0.6.3-beta2 I can use the file browser existing inside Guayadeque, select an aac file play it of course this aac file doesn't happear in the library
> 
> Didn't you try to update the library to access the AAC from mediaviewer? You can try this, it should be added to the library without issues.

I did try to update the library but it doesn't appear inside the library :  I can only find the aac file inside the source panel (when I click to show the sources)
> 
> > When I select the aac file playing and try to edit the tags of this track, the tag editor is grayed : that's what is wanted ! This doesn't happen for mp3 m4a ogg etc for them we can edit their tag.
> 
> Great! In my tests with Taglib 2, it appears editable and I was able to edit without issues (I checked with Kid3 if the tag was updated!).

About the grayed tag editor when trying to tag an aac file : it's a progress in 0.6.3-beta 2 since inside 0.6.2  it was not grayed and we might think it was possible but immediately lost

**Comment by thothix** [comment]
One more detail: It is not only AAC files that will work that way, any file that not support tagging will appears disabled in the track editor.

**Comment by filochard** [comment]
> One more detail: It is not only AAC files that will work that way, any file that not support tagging will appears disabled in the track editor.

Except aac we can see tags   for  :  flac, m4a, mp3, musepack, ogg, opus, wav, and wma...  inside Guayadeque
NB I don't remember if I have edited these tags inside Guayadeque,  or with musicbrainz or kid3 before importing them in my music directory where Guayadeque looks for the files to eventually add to the library

**Comment by filochard** [comment]
last tests with 0.6.3-beta 2
I can edit tags for flac, m4a, mp3, musepack, ogg, opus, wav, and wma
for aac the tag editor is grayed
it's OK 
Importing files creates always problems in the tree of the music directory if the tag doesn't contain album-artist-name , artist-name, album-name, and title 
in this case it creates lots of sub-folders such as /inconnu/inconnu/inconnu/tagged-title as an example

Besides this

I have tried to build guayadeque for Mageia9 upon taglib2 with mock  : it is possible but !
Taglib1 is needed by lots of other packages that would need to be rebuilt for taglib2 if I update it ... no test possible inside Mageia9 
Cauldron is really unstable now and I dare not to use it 
(updating rpm to 4.20 inside Cauldron has induced lots of build failures needing to correct lots of spec files before a mass rebuild)

**Comment by thothix** [comment]
I got Cauldron running in a VM. I upgraded Mageia 9 VM to Cauldron. I had some issues to upgrade it but I got it.
I tested Guayadeque and AAC tag is working in Cauldron. 
I found out a bug but I need to investigate further. It's something related to the file name (long file name with accents)!
 I wasn't able to tag a AAC file and it worked after I renamed it!

**Comment by filochard** [comment]
> I got Cauldron running in a VM. I upgraded Mageia 9 VM to Cauldron. I had some issues to upgrade it but I got it. I tested Guayadeque and AAC tag is working in Cauldron. I found out a bug but I need to investigate further. It's something related to the file name (long file name with accents)! I wasn't able to tag a AAC file and it worked after I renamed it!

Nice if you could do it inside a VM

About files names : I never use accented characters nor long names : some programs can't read them...
For music files sometimes I have to copy them to NTFS external devices and such kind of file names are not accepted.
Besides these accented characters are accepted in tags, particularly for embedded Lyrics that can be written in any language 
(Greek Ukrainian Arab  Hindi Sanskrit Thai etc... with a transliteration to pronounce them and a translation)
These Lyrics can be read by some good music players such as Guayadeque  ;o)

**Comment by thothix** [comment]
I implemented an enhancement in the covers scan, so now, besides it find covers with the "Words to detect covers" in collection preferences, it finds covers that has the album name (by default) (beta 3)

Example for an album named "Mind Body & Soul" with the following images in the folder:
- cover.jpg
- 2004_Mind Body & Soul.jpg

it will use "cover.jpg"

If it has only the  "2004_Mind Body & Soul.jpg" or  "Mind Body & Soul.jpg", it 'will use it.

**Comment by filochard** [comment]
Hi Tiago
I have tested beta3 with a micro directory (/home/myname/testmusic/)containing  one file of each format 
aac flac, m4a, mp3, musepack, ogg, opus, wav, and wma
all of them were previously tagged
For musepack, wav, those tags are lost when I try to read them with Guayadeque 
Nevertheless I can edit these tags inside Guayadeque (they are apparently preserved in the guayadeque.db)
and these tags are rewritten for the wav and musepack files inside the directory

With these tags I can download a cover for each of the albums but I don't understand where these covers are stored : is it in this quayadeque.db ?
The only cover I can find in the /home/myname/testmusic/ directory besides the music files is the last cover.jpg downloaded

**Comment by thothix** [comment]
> Hi Tiago I have tested beta3 with a micro directory (/home/myname/testmusic/)containing one file of each format aac flac, m4a, mp3, musepack, ogg, opus, wav, and wma all of them were previously tagged For musepack, wav, those tags are lost when I try to read them with Guayadeque Nevertheless I can edit these tags inside Guayadeque (they are apparently preserved in the guayadeque.db) and these tags are rewritten for the wav and musepack files inside the directory
> 
> With these tags I can download a cover for each of the albums but I don't understand where these covers are stored : is it in this quayadeque.db ? The only cover I can find in the /home/myname/testmusic/ directory besides the music files is the last cover.jpg downloaded

In the database are stored only the path to the cover file and a thumbnail that is generated during the scan. The cover need to be in the audio file directory or to be embedded inside the audio file  (to be found during the scan) or you can manually select one from other location through the "select cover" option.

**Comment by thothix** [comment]
The tag data are always written in the database, independent of the file format or support. This is a source of problems because we don't know if the tag was actually written in the audio file or not. You ALWAYS need to check the file tag to know if it was written in the tests. 
A cover will be found if it was defined from a external file but if it is embedded it has the same problem.

**Comment by filochard** [comment]
Hi Tiago
Some screenshots are better than long speeches 
I did a test with beta3  
1)  I used for this a little folder (testmusic) with one file of each format

![treetest](https://github.com/user-attachments/assets/bde21c75-dabd-47c2-8a83-f542fa8dee43)

2) Then I had to write some lost tags in guayadeque ... this modified 2 tags in the folder (NB this screenshot shows additionally the last downloaded cover.jpg overwritten 6 times)

![directory](https://github.com/user-attachments/assets/31105b4f-7d06-4ab6-b92b-f86542c0bace)

3) Then I downloaded covers

![covers](https://github.com/user-attachments/assets/8f336c27-1f11-421d-9d56-7f4a05f7d3e2)

4) the layout showed this covers

![layout](https://github.com/user-attachments/assets/cab61bb7-9e28-40d4-895e-2c68c74cffc4)

5) I chosed a track and played it.... but it displayed the last cover downloaded : cover.jpg

![playing-layout](https://github.com/user-attachments/assets/78ccb818-707d-4397-b2c5-6015d2e98c39)

6) After having played the 8 tracks, I quit Guayadeque and start it again all the covers are lost

![last-layout](https://github.com/user-attachments/assets/661f27d6-eaad-476d-b7d2-0b7299c644b1)

To summarize the problem : 
The downloaded covers appears correctly until one track is played

Downloaded cover files should have the name of the album and not be named cover.jpg : 
this is OK for one album having its own folder  
but if a folder contains a compilation of tracks from several albums that is not good : we need them several allbum-name.jpg

**Comment by thothix** [comment]
Hi,
The library (not only Guayadeque, the Music Players that I know that has a library) are made to have only 1  album per directory (the import files try to separate the files this way also - artist/album at least). If not you don't have a library, you have a bunch of music files ;-) ! But if you do this, you need to name each downloaded cover with a unique name on the directory. The cover has a fixed name by default (not only to download but to search in the scanning as well)  because of  this.

**Comment by thothix** [comment]
I like to name my covers with the album name but, as most music players (includeing Guayadeque), name them as "cover" or "folder"or other standard name, I keep them as "cover". I don't like to name them this way because, if I copy a cover to another place, I don't know to what album this cover belongs to, all albums' covers has the same name!
Now this issue is solved.

Maybe we can change the default name when downloading from "cover.jpg" to the album name ".jpg" (if the file has an album name) but I don't know if it's a good thing to break the standards

**Comment by filochard** [comment]
What was strange for me was that after having updated the covers (look inside https://github.com/thothix/guayadeque/issues/33#issuecomment-2558106399  paragraph 3) & 4)   ) one cover is lost for the first played track paragraph 5)  but all disappear after aquit and restart.

Apparently that has no solution for such a compilation

You said : 
"I like to name my covers with the album name"
What I usually do for each of my album folders is to copy inside this folder the cover.jpg downloaded as album_name.jpg : both of them may coexist... 
(sometimes, when Guayadeque has not found a cover inside Apple or Last.Fm, I search elsewhere this cover and copy it inside the album folder and chose it then with Guayadeque and it is preserved)

Besides this, what I usually do for some folders containing several tagged tracks from different artists and albums:  inside Guayadeque I can choose a  specified album_track jpg or png for each of these albums and this is correctly set then)

**Comment by thothix** [comment]
> What was strange for me was that after having updated the covers (look inside [#33 (comment)](https://github.com/thothix/guayadeque/issues/33#issuecomment-2558106399) paragraph 3) & 4) ) one cover is lost for the first played track paragraph 5) but all disappear after aquit and restart.
> 

It's because you set one by one and the albums was set correctly and remained in memory (in the albums list) but all pointing to "cover.jpg". When you restart Guayadeque (or update the covers) all albums covers are reloaded and all albums covers reload the same file/cover on disk.

> Apparently that has no solution for such a compilation
> 

As I said, the only solution is set a different cover name to each song/album in the directory. How can it have 2 albums with different covers with the same name?

> You said : "I like to name my covers with the album name" What I usually do for each of my album folders is to copy inside this folder the cover.jpg downloaded as album_name.jpg : both of them may coexist... (sometimes, when Guayadeque has not found a cover inside Apple or Last.Fm, I search elsewhere this cover and copy it inside the album folder and chose it then with Guayadeque and it is preserved)
> 

Yes, fine. One note: Before beta3 the album_name.jpg was ignored by Guayadeque.

> Besides this, what I usually do for some folders containing several tagged tracks from different artists and albums: inside Guayadeque I can choose a specified album_track jpg or png for each of these albums and this is correctly set then)

Yes, sure. That's what I said.

**Comment by thothix** [comment]
I'm going to enable the library to import .DSF files but I need to work on tagging support (only for Taglib 2).
I'll try to do it still in 0.6.3. If I'm not able to finish it in 0.6.3, the track editor will be disabled for .DSF editing, like for .AAC files.

**Comment by thothix** [comment]
v0.6.3 released.

**Comment by filochard** [comment]
Thanks Tiago
I'm gonna package it and test it
